### PR TITLE
refactor(api,worker): a platform can have many plugins

### PR DESCRIPTION
**Pull request by yesnault**
part of ref: #3551
Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

File: engine/api/plugin/dao.go [hunk 1]
```diff
@@ -72,10 +73,30 @@ func LoadByName(db gorp.SqlExecutor, name string) (*sdk.GRPCPlugin, error) {
...
 	p := sdk.GRPCPlugin(m)
 	return &p, nil
 }
 
+// LoadByPlatformModelIDAndType loads all plugins associated to a platform model id
+func LoadByPlatformModelIDAndType(db gorp.SqlExecutor, platformModelID int64, typePlugin string) (*sdk.GRPCPlugin, error) {
+	m := grpcPlugin{}
+	if err := db.SelectOne(&m, "SELECT * FROM grpc_plugin where platform_model_id = $1 and type = $2", platformModelID, typePlugin); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, sdk.ErrNotFound
```
Comment (richardlt) [inline engine/api/plugin/dao.go]:
```suggestion
			return nil, sdk.WithStack(sdk.ErrNotFound)
```

File: engine/api/workflow/execute_node_run.go [hunk 2]
```diff
@@ -453,12 +453,12 @@ func addJobsToQueue(ctx context.Context, db gorp.SqlExecutor, stage *sdk.Stage,
 }
 
 func getPlatformPluginBinaries(db gorp.SqlExecutor, runContext nodeRunContext) ([]sdk.GRPCPluginBinary, error) {
-	if runContext.ProjectPlatform.Model.PluginName != "" {
-		p, err := plugin.LoadByName(db, runContext.ProjectPlatform.Model.PluginName)
+	if runContext.ProjectPlatform.Model.ID > 0 {
+		plugin, err := plugin.LoadByPlatformModelIDAndType(db, runContext.ProjectPlatform.Model.ID, sdk.GRPCPluginDeploymentPlatform)
 		if err != nil {
-			return nil, sdk.WrapError(sdk.ErrWorkflowNodeNotFound, "getPlatformPluginBinaries> Cannot find plugin %s, %v", runContext.ProjectPlatform.Model.PluginName, err)
+			return nil, sdk.WrapError(sdk.ErrWorkflowNodeNotFound, "getPlatformPluginBinaries> Cannot find plugin for platform model id %d, %v", runContext.ProjectPlatform.Model.ID, err)
```
Comment (fsamin) [inline engine/api/workflow/execute_node_run.go]: NewError

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#8136.0 ✘  
* Build minimal
  * cdsctl linux ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔ 
* Build other os/arch
  * cdsctl darwin -   
  * cdsctl freebsd windows amd64 -   
  * cdsctl openbsd amd64 -   
  * Doc Build -   
  * Engine Build darwin amd64 -   
  * Engine Build freebsd amd64 -   
  * Engine Build linux arm -   
  * Engine Build openbsd amd64 -   
  * Engine Build windows amd64 -   
  * Plugins darwin amd64 -   
  * Plugins freebsd -   
  * Plugins linux amd64 -   
  * Plugins linux arm -   
  * Plugins openbsd -   
  * Plugins windows amd64 -   
  * Worker darwin / windows / freebsd / openbsd -   
  * Worker linux arm -   
* Package
  * Debian Engine Package ✔ 
  * Docker Engine Package ✘

File: engine/api/workflow/execute_node_run.go (hunk 2 above)
Comment (yesnault) [reply engine/api/workflow/execute_node_run.go]: done

**Pull request merged**